metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/SequenceGeneration.java

```diff
@@ -127,6 +131,48 @@ public PartitionSequenceIds newPartitionSequenceIds(final int size) {
         }
     }
 
+    /**
+     * Returns the current sequence ids and increments the sequence ids by the given <code>size</code>.
+     * @param size number of records getting inserted
+     * @param sequenceParamName  the sequence Parameter Name
+     * @return current sequence ids
+     */
+    @Transactional(propagation = Propagation.REQUIRES_NEW, isolation = Isolation.REPEATABLE_READ)
+    public Long newPartitionSequenceIdByName(final int size, final String sequenceParamName) {
+        Long result = null;
+        try {
+            result = jdbcTemplate.queryForObject(SQL.SEQUENCE_NEXT_VAL_BYNAME,
+                new Object[]{sequenceParamName}, Long.class);
+        } catch (DataAccessException e) {
+            log.warn("Failed getting the sequence ids for partition", e);
+        } catch (Exception e) {
+            throw new IllegalStateException("Failed retrieving the sequence numbers.");
+        }
+
+        try {
+            final List<Object[]> insertValues = Lists.newArrayList();
+            final List<Object[]> updateValues = Lists.newArrayList();
+            if (result == null) {
+                result = 1L;
+                insertValues.add(new Object[] {result + size, sequenceParamName });
+            } else {
+                updateValues.add(new Object[] {result + size, sequenceParamName });
+            }
+
+            if (!insertValues.isEmpty()) {
+                jdbcTemplate.batchUpdate(SQL.SEQUENCE_INSERT_VAL, insertValues,
```

Comment:
Use an simple update instead of a batch update. You can also get rid of the list of values